File: media/js/firefox/mobile/mobile.js
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

(function() {
    'use strict';

    var modal = document.getElementById('modal');
    var mobileButtons = document.querySelectorAll('.js-mobile');
    var mobileContent = document.getElementById('modal-download-firefox');

    for(var i = 0; i < mobileButtons.length; i++) {
        mobileButtons[i].addEventListener('click', function() {
            mobileContent.style.display = 'block';
            Mzp.Modal.createModal(this, modal, {
                closeText: window.Mozilla.Utils.trans('global-close'),
                onDestroy: function() {
                    mobileContent.style.display = 'none';
                }
            });
        });
    }

    // initialize send to device widget
    var form = new Mozilla.SendToDevice();
    form.init();

})();

File: tests/pages/firefox/mobile.py
# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at http://mozilla.org/MPL/2.0/.

from selenium.webdriver.common.by import By

from pages.base import BasePage
from pages.regions.send_to_device import SendToDevice


class FirefoxMobilePage(BasePage):

    URL_TEMPLATE = '/{locale}/firefox/mobile/'

    _get_firefox_header_button_locator = (By.ID, 'get-firefox')
    _get_firefox_qr_code_locator = (By.ID, 'firefox-qr')

    @property
    def send_to_device(self):
        return SendToDevice(self)

    @property
    def is_firefox_qr_code_displayed(self):
        return self.is_element_displayed(*self._get_firefox_qr_code_locator)
